# HG changeset patch
# Parent 59409fb3805f44217b7d1037f465ad8e20a09f55
# User Frédéric Wang <fred.wang@free.fr>
MathML Operator Dictionary: add new fence, accent and largeop entries (bug 534970). r=karlt

diff --git a/layout/mathml/mathfont.properties b/layout/mathml/mathfont.properties
--- a/layout/mathml/mathfont.properties
+++ b/layout/mathml/mathfont.properties
@@ -166,39 +166,48 @@ operator.\u005E.postfix = lspace:0 rspac
 operator.\u005E.infix = lspace:1 rspace:1 direction:horizontal # ^
 operator.\u005F.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # _ low line
 operator.\u005F.infix = lspace:2 rspace:2 stretchy direction:horizontal # _
 operator.\u0060.postfix = lspace:0 rspace:0 accent # &DiacriticalGrave;
 operator.\u007B.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # {
 operator.\u007C.infix = lspace:0 rspace:0 stretchy symmetric direction:vertical # &VerticalLine; |
 operator.\u007C.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # |
 operator.\u007C.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # |
-operator.\u007C\u007C.infix = lspace:4 rspace:4 # ||
+operator.\u007C\u007C.infix = lspace:4 rspace:4 direction:vertical # ||
+operator.\u007C\u007C.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # multiple character operator: ||
+operator.\u007C\u007C.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # multiple character operator: ||
+operator.\u007C\u007C\u007C.infix = lspace:2 rspace:2 stretchy fence symmetric direction:vertical # multiple character operator: |||
+operator.\u007C\u007C\u007C.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # multiple character operator: |||
+operator.\u007C\u007C\u007C.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # multiple character operator: |||
 operator.\u007D.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # }
 operator.\u007E.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # ~ tilde
 operator.\u00A8.postfix = lspace:0 rspace:0 accent # &DoubleDot;
 operator.\u00AC.prefix = lspace:2 rspace:1 # not sign
 operator.\u00AF.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &OverBar;
 operator.\u00B0.postfix = lspace:0 rspace:0 # degree sign
 operator.\u00B1.infix = lspace:4 rspace:4 # &PlusMinus;
 operator.\u00B1.prefix = lspace:0 rspace:1 # &PlusMinus;
 operator.\u00B4.postfix = lspace:0 rspace:0 accent # &DiacriticalAcute;
 operator.\u00B7.infix = lspace:4 rspace:4 # &CenterDot;
 operator.\u00B8.postfix = lspace:0 rspace:0 accent # &Cedilla;
 operator.\u00D7.infix = lspace:4 rspace:4 # multiplication sign
 operator.\u00F7.infix = lspace:4 rspace:4 # division sign
 operator.\u02C6.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # modifier letter circumflex accent
 operator.\u02C7.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &Hacek; caron
+operator.\u02C9.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # modifier letter macron
 operator.\u02CA.postfix = lspace:0 rspace:0 accent # modifier letter acute accent
 operator.\u02CB.postfix = lspace:0 rspace:0 accent # modifier letter grave accent
+operator.\u02CD.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # modifier letter low macron
 operator.\u02D8.postfix = lspace:0 rspace:0 accent # &Breve;
 operator.\u02D9.postfix = lspace:0 rspace:0 accent # &DiacriticalDot;
 operator.\u02DA.postfix = lspace:0 rspace:0 accent # ring above
 operator.\u02DC.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &DiacriticalTilde; small tilde
 operator.\u02DD.postfix = lspace:0 rspace:0 accent # &DiacriticalDoubleAcute;
+operator.\u02F7.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # modifier letter low tilde
+operator.\u0302.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # combining circumflex accent
 operator.\u0311.postfix = lspace:0 rspace:0 accent # &DownBreve;
 operator.\u03F6.infix = lspace:5 rspace:5 # greek reversed lunate epsilon symbol
 operator.\u2016.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &Vert; &Verbar;
 operator.\u2016.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &Vert; &Verbar;
 operator.\u2018.prefix = lspace:0 rspace:0 fence # &OpenCurlyQuote;
 operator.\u2019.postfix = lspace:0 rspace:0 fence # &CloseCurlyQuote;
 operator.\u201C.prefix = lspace:0 rspace:0 fence # &OpenCurlyDoubleQuote;
 operator.\u201D.postfix = lspace:0 rspace:0 fence # &CloseCurlyDoubleQuote;
@@ -593,16 +602,18 @@ operator.\u2309.postfix = lspace:0 rspac
 operator.\u230A.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &LeftFloor;
 operator.\u230B.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &RightFloor;
 operator.\u23B4.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &OverBracket;
 operator.\u23B5.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &UnderBracket;
 operator.\u23DC.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &OverParenthesis; (Unicode)
 operator.\u23DD.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &UnderParenthesis; (Unicode)
 operator.\u23DE.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &OverBrace; (Unicode)
 operator.\u23DF.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # &UnderBrace; (Unicode)
+operator.\u23E0.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # top tortoise shell bracket
+operator.\u23E1.postfix = lspace:0 rspace:0 stretchy accent direction:horizontal # bottom tortoise shell bracket
 operator.\u25A0.infix = lspace:3 rspace:3 # black square
 operator.\u25A1.infix = lspace:3 rspace:3 # white square
 operator.\u25AA.infix = lspace:3 rspace:3 # black small square
 operator.\u25AB.infix = lspace:3 rspace:3 # white small square
 operator.\u25AD.infix = lspace:3 rspace:3 # white rectangle
 operator.\u25AE.infix = lspace:3 rspace:3 # black vertical rectangle
 operator.\u25AF.infix = lspace:3 rspace:3 # white vertical rectangle
 operator.\u25B0.infix = lspace:3 rspace:3 # black parallelogram
@@ -635,26 +646,39 @@ operator.\u25CE.infix = lspace:4 rspace:
 operator.\u25CF.infix = lspace:4 rspace:4 # black circle
 operator.\u25D6.infix = lspace:4 rspace:4 # left half black circle
 operator.\u25D7.infix = lspace:4 rspace:4 # right half black circle
 operator.\u25E6.infix = lspace:4 rspace:4 # white bullet
 operator.\u266D.postfix = lspace:0 rspace:2 # music flat sign
 operator.\u266E.postfix = lspace:0 rspace:2 # music natural sign
 operator.\u266F.postfix = lspace:0 rspace:2 # music sharp sign
 operator.\u2758.infix = lspace:5 rspace:5 stretchy symmetric direction:vertical # &VerticalSeparator;
+operator.\u2772.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # light left tortoise shell bracket ornament
+operator.\u2773.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # light right tortoise shell bracket ornament
 operator.\u27E6.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &LeftDoubleBracket;
 operator.\u27E7.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &RightDoubleBracket;
 operator.\u27E8.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &LeftAngleBracket;
 operator.\u27E9.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # &RightAngleBracket;
+operator.\u27EA.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical left double angle bracket
+operator.\u27EB.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical right double angle bracket
+operator.\u27EC.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical left white tortoise shell bracket
+operator.\u27ED.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical right white tortoise shell bracket
+operator.\u27EE.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical left flattened parenthesis
+operator.\u27EF.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # mathematical right flattened parenthesis
 operator.\u27F5.infix = lspace:2 rspace:2 stretchy direction:horizontal # &LongLeftArrow;
 operator.\u27F6.infix = lspace:2 rspace:2 stretchy direction:horizontal # &LongRightArrow;
 operator.\u27F7.infix = lspace:2 rspace:2 stretchy direction:horizontal # &LongLeftRightArrow;
 operator.\u27F8.infix = lspace:2 rspace:2 stretchy direction:horizontal # &DoubleLongLeftArrow;
 operator.\u27F9.infix = lspace:2 rspace:2 stretchy direction:horizontal # &DoubleLongRightArrow;
 operator.\u27FA.infix = lspace:2 rspace:2 stretchy direction:horizontal # &DoubleLongLeftRightArrow;
+operator.\u27FB.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # long leftwards arrow from bar
+operator.\u27FC.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # long rightwards arrow from bar
+operator.\u27FD.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # long leftwards double arrow from bar
+operator.\u27FE.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # long rightwards double arrow from bar
+operator.\u27FF.infix = lspace:5 rspace:5 stretchy accent direction:horizontal # long rightwards squiggle arrow
 operator.\u2900.infix = lspace:5 rspace:5 accent # rightwards two-headed arrow with vertical stroke
 operator.\u2901.infix = lspace:5 rspace:5 accent # rightwards two-headed arrow with double vertical stroke
 operator.\u2902.infix = lspace:5 rspace:5 accent # leftwards double arrow with vertical stroke
 operator.\u2903.infix = lspace:5 rspace:5 accent # rightwards double arrow with vertical stroke
 operator.\u2904.infix = lspace:5 rspace:5 accent # left right double arrow with vertical stroke
 operator.\u2905.infix = lspace:5 rspace:5 accent # rightwards two-headed arrow from bar
 operator.\u2906.infix = lspace:5 rspace:5 accent # leftwards double arrow from bar
 operator.\u2907.infix = lspace:5 rspace:5 accent # rightwards double arrow from bar
@@ -771,16 +795,18 @@ operator.\u2977.infix = lspace:5 rspace:
 operator.\u2978.infix = lspace:5 rspace:5 accent # greater-than above rightwards arrow
 operator.\u2979.infix = lspace:5 rspace:5 accent # subset above rightwards arrow
 operator.\u297A.infix = lspace:5 rspace:5 accent # leftwards arrow through subset
 operator.\u297B.infix = lspace:5 rspace:5 accent # superset above leftwards arrow
 operator.\u297C.infix = lspace:5 rspace:5 accent # left fish tail
 operator.\u297D.infix = lspace:5 rspace:5 accent # right fish tail
 operator.\u297E.infix = lspace:5 rspace:5 # up fish tail
 operator.\u297F.infix = lspace:5 rspace:5 # down fish tail
+operator.\u2980.prefix = lspace:0 rspace:0 stretchy fence direction:vertical # triple direction:vertical bar delimiter
+operator.\u2980.postfix = lspace:0 rspace:0 stretchy fence direction:vertical # triple direction:vertical bar delimiter
 operator.\u2981.infix = lspace:3 rspace:3 # z notation spot
 operator.\u2982.infix = lspace:3 rspace:3 # z notation type colon
 operator.\u2983.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # left white curly bracket
 operator.\u2984.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # right white curly bracket
 operator.\u2985.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # left white parenthesis
 operator.\u2986.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # right white parenthesis
 operator.\u2987.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # z notation left image bracket
 operator.\u2988.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # z notation right image bracket
@@ -902,18 +928,24 @@ operator.\u29FA.infix = lspace:3 rspace:
 operator.\u29FB.infix = lspace:3 rspace:3 # triple plus
 operator.\u29FC.prefix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # left-pointing curved angle bracket
 operator.\u29FD.postfix = lspace:0 rspace:0 stretchy fence symmetric direction:vertical # right-pointing curved angle bracket
 operator.\u29FE.infix = lspace:4 rspace:4 # tiny
 operator.\u29FF.infix = lspace:4 rspace:4 # miny
 operator.\u2A00.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # &bigodot;
 operator.\u2A01.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # &bigoplus;
 operator.\u2A02.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # &bigotimes;
+operator.\u2A03.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # n-ary union operator with dot
 operator.\u2A04.prefix = lspace:1 rspace:2 largeop movablelimits symmetric direction:vertical # &biguplus;
+operator.\u2A05.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # n-ary square intersection operator
 operator.\u2A06.prefix = lspace:1 rspace:2 largeop movablelimits symmetric direction:vertical # &bigsqcup;
+operator.\u2A07.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # two logical and operator
+operator.\u2A08.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # two logical or operator
+operator.\u2A09.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # n-ary times operator
+operator.\u2A0A.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # modulo two sum
 operator.\u2A0B.prefix = lspace:1 rspace:2 largeop symmetric direction:vertical integral # summation with integral
 operator.\u2A0C.prefix = lspace:0 rspace:1 largeop symmetric direction:vertical integral # quadruple integral operator
 operator.\u2A0D.prefix = lspace:1 rspace:2 largeop symmetric direction:vertical integral # finite part integral
 operator.\u2A0E.prefix = lspace:1 rspace:2 largeop symmetric direction:vertical integral # integral with double stroke
 operator.\u2A0F.prefix = lspace:1 rspace:2 largeop symmetric direction:vertical integral # integral average with slash
 operator.\u2A10.prefix = lspace:1 rspace:2 largeop movablelimits symmetric direction:vertical integral # circulation function
 operator.\u2A11.prefix = lspace:1 rspace:2 largeop movablelimits symmetric direction:vertical integral # anticlockwise integration
 operator.\u2A12.prefix = lspace:1 rspace:2 largeop movablelimits symmetric direction:vertical integral # line integration with rectangular path around pole
@@ -1151,18 +1183,20 @@ operator.\u2AF3.infix = lspace:5 rspace:
 operator.\u2AF4.infix = lspace:4 rspace:4 # triple vertical bar binary relation
 operator.\u2AF5.infix = lspace:4 rspace:4 # triple vertical bar with horizontal stroke
 operator.\u2AF6.infix = lspace:4 rspace:4 # triple colon operator
 operator.\u2AF7.infix = lspace:5 rspace:5 # triple nested less-than
 operator.\u2AF8.infix = lspace:5 rspace:5 # triple nested greater-than
 operator.\u2AF9.infix = lspace:5 rspace:5 # double-line slanted less-than or equal to
 operator.\u2AFA.infix = lspace:5 rspace:5 # double-line slanted greater-than or equal to
 operator.\u2AFB.infix = lspace:4 rspace:4 # triple solidus binary relation
+operator.\u2AFC.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # large triple vertical bar operator
 operator.\u2AFD.infix = lspace:4 rspace:4 # double solidus operator
 operator.\u2AFE.infix = lspace:3 rspace:3 # white vertical bar
+operator.\u2AFF.prefix = lspace:1 rspace:2 largeop movablelimits symmetric # n-ary white vertical bar
 
 # Entries below are not part of the official MathML dictionary
 
 operator.\u0021.prefix = lspace:0 rspace:5 # !
 operator.\u0026.infix = lspace:5 rspace:5 # &amp;
 operator.\u0026.prefix = lspace:0 rspace:5 # &amp;
 operator.\u002B\u002B.prefix = lspace:0 rspace:2 # ++
 operator.\u002D\u002D.prefix = lspace:0 rspace:2 # --
